common/buildcraft/factory/tile/TileTank.java

```diff
@@ -0,0 +1,87 @@
+package buildcraft.factory.tile;
+
+import buildcraft.api.tiles.IDebuggable;
+import buildcraft.lib.fluids.SingleUseTank;
+import buildcraft.lib.tile.TileBC_Neptune;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ITickable;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+import net.minecraftforge.fml.relauncher.Side;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.List;
+
+public class TileTank extends TileBC_Neptune implements ITickable, IDebuggable {
+    public static final int NET_TANK = 10;
+
+    public SingleUseTank tank = new SingleUseTank("tank", 16000, this);
```

Comment:
You don't want to use SingleUseTank here- that only allows for a single type of fluid to be in the tank, ever. Which is fine for the pump (we don't want to start by pumping out oil and then start to pump out water) the tank just needs to accept any fluid.